sandbox/routes/patient/patch-patient.js

```diff
@@ -0,0 +1,78 @@
+const Boom = require('boom')
+const fs = require('fs')
+const nhsNumberHelper = require('../../helpers/nhs-number-helper')
+const fhirHelper = require('../../helpers/fhir-helper')
+const jsonpatch = require('fast-json-patch')
+
+const EXAMPLE_PATIENT = JSON.parse(fs.readFileSync('mocks/Patient-Jane-Smith.json'))
+
+module.exports = [
+    {
+        /*
+            Patient partial update
+            Behaviour:
+            * No If-Match Header: 400 DONE
+            * If-Match header does not match resource latest: 412 Precondition Failed DONE
+            * Invalid request body: 400(?)
+            * Non-existant patient: 404 DONE
+            * Invalid NHS Number: 400 DONE
+            * Unset/invalid Content-Type: 415 Unsupported Media Type DONE
+        */
+        method: 'PATCH',
+        path: '/Patient/{nhsNumber}',
+        handler: (request, h) => {
+            nhsNumberHelper.checkNhsNumber(request)
+
+            // Check If-Match header exists
+            // TODO: Return different error where If-Match header is incorrect format
+            if (
+                !request.headers["if-match"] ||
```

Comment:
These conditions are getting a little unwieldy; it would be worth breaking them out into a predicate function.